### PR TITLE
[REFACTOR] - Move this project to be a ruby gem.

### PR DESCRIPTION
**Pull request by yossi-eynav**

### CODE REVIEW

File: rollout_service.gemspec [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# coding: utf-8
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'rollout_service/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = "rollout_service"
+  spec.version       = RolloutService::VERSION
+  spec.authors       = ["Yossi Eynav"]
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Fiverr

File: rollout_service.gemspec [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  spec.email         = ["yossi.eynav@gmail.com"]
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: dev@fiverr.com

File: rollout_service.gemspec [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  spec.summary       = "This gem exposes rollout gem API"
+  spec.homepage      = "https://www.fiverr.com"
+
+  # Prevent pushing this gem to RubyGems.org. To allow pushes either set the 'allowed_push_host'
+  # to allow pushing to a single host or delete this section to allow pushing to any host.
+  if spec.respond_to?(:metadata)
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Remove

File: rollout_service.gemspec [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    spec.metadata['allowed_push_host'] = "http://mygemserver.com"
+  else
+    raise "RubyGems 2.0 or newer is required to protect against public gem pushes."
+  end
+
+  spec.files         = `git ls-files -z`.split("\x0").reject do |f|
+    f.match(%r{^(test|spec|features)/})
+  end
+  spec.bindir        = "exe"
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Remove + delete bin, exe dir

File: rollout_service.gemspec [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Remove

File: rollout_service.gemspec [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  spec.require_paths = ["lib"]
+
+  spec.add_dependency 'grape'
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Version

File: rollout_service.gemspec [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  spec.add_dependency 'grape-entity', '~> 0.5.0'
+  spec.add_dependency 'rollout'
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Version

File: rollout_service.gemspec [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  spec.add_dependency 'redis'
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Version

File: rollout_service.gemspec [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  spec.add_dependency 'require_all'
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Version

File: rollout_service.gemspec [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  spec.add_dependency 'activesupport', '~> 5.0'
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: Okay per se, but you won't be able to use other fiverr gems.

File: rollout_service.gemspec [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  spec.add_dependency 'active_attr', '~> 0.9.0'
```
Comment (oded-zahavi) [inline rollout_service.gemspec]: ~> 0.9

File: lib/rollout_service/models/user.rb [hunk 12]
```diff
@@ -0,0 +1,12 @@
+module RolloutService
```
Comment (oded-zahavi) [inline lib/rollout_service/models/user.rb:1]: Not in models. This is not a model!

File: lib/rollout_service/models/feature.rb [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+        Config::rollout.delete(self.name)
+      end
+
+      private
+
+      def set_history_attribute
+        last_record = history.last
+        return if last_record.present? && last_record[:percentage] == self.percentage
+
+        self.history << {
```
Comment (oded-zahavi) [inline lib/rollout_service/models/feature.rb:97]: Creat object, add to history, and return the object. Better then adding and getting last.

**Review by oded-zahavi — CHANGES_REQUESTED** [review]
Comments inline

File: lib/rollout_service/models/user.rb (hunk 12 above)
Comment (yossi-eynav) [reply lib/rollout_service/models/user.rb:1]: where do you suggest to put it?

File: lib/rollout_service/models/feature.rb (hunk 13 above)
Comment (yossi-eynav) [reply lib/rollout_service/models/feature.rb:97]: can you please explain further?

**Pull request merged**